code/modules/mob/mob_movement.dm

```diff
@@ -148,7 +148,10 @@
 		return
 	if(!mob)
 		return // Moved here to avoid nullrefs below
-	return mob.SelfMove(direction)
+	. = mob.SelfMove(direction)
+	if(.)
+		next_move_dir_add = 0
+		next_move_dir_sub = 0
```

Comment:
I don't think this does anything.

Reply:
Please don't mark my comments as resolved unless you have addressed them. In fact, don't mark them as resolved at all, ever: reply to them instead.